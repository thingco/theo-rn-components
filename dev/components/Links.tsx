import React from "react";
import { TouchableOpacity, TextStyle, StyleProp } from "react-native";
import { Text } from "./Base";

import s from "./Styles";

type LinkProps = {
  children: React.ReactNode;
  onPress: () => void;
  style?: StyleProp<TextStyle>;
};

/** External links and groups of links, styled like standard HTML w/underline: */
const Link: React.FC<LinkProps> = (props: LinkProps) => (
  <TouchableOpacity style={props.style} onPress={props.onPress}>
    <Text style={[s.white, s.text_sm, s.underline]}>{props.children}</Text>
  </TouchableOpacity>
);

export { Link };
