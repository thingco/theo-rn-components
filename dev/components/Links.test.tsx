import React from "react";
import renderer from "react-test-renderer";

import { Link } from "./Links";

test("Link renders correctly", () => {
  const tree = renderer
    .create(<Link onPress={() => void 0}>I'm a link!</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
