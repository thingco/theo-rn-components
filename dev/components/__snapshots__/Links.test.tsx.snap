// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Bariol",
        },
        Object {
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#FFFFFF",
          },
          Object {
            "fontSize": 14,
          },
          Object {
            "textDecorationLine": "underline",
          },
        ],
      ]
    }
  >
    I'm a link!
  </Text>
</View>
`;
